test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -708,28 +709,44 @@ var _ = ginkgo.Describe("ClusterQueue controller", ginkgo.Ordered, ginkgo.Contin
 
 	ginkgo.When("Deleting clusterQueues", func() {
 		var (
-			cq *kueue.ClusterQueue
-			lq *kueue.LocalQueue
+			cq    *kueue.ClusterQueue
+			lq    *kueue.LocalQueue
+			check *kueue.AdmissionCheck
 		)
 
 		ginkgo.BeforeEach(func() {
-			cq = testing.MakeClusterQueue("foo-cq").Obj()
+			check = testing.MakeAdmissionCheck("check").ControllerName("check-controller").Obj()
+			gomega.Expect(k8sClient.Create(ctx, check)).To(gomega.Succeed())
+
+			cq = testing.MakeClusterQueue("foo-cq").AdmissionChecks(check.Name).Obj()
 			lq = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(cq.Name).Obj()
 			gomega.Expect(k8sClient.Create(ctx, lq)).To(gomega.Succeed())
 			gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
 		})
 
+		ginkgo.AfterEach(func() {
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check, true)
+		})
+
 		ginkgo.It("Should delete clusterQueues successfully when no admitted workloads are running", func() {
 			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cq, true)
 		})
 
 		ginkgo.It("Should be stuck in termination until admitted workloads finished running", func() {
+			util.SetAdmissionCheckActive(ctx, k8sClient, check, metav1.ConditionTrue)
 			util.ExpectClusterQueueStatusMetric(cq, metrics.CQStatusActive)
 
 			ginkgo.By("Admit workload")
 			wl := testing.MakeWorkload("workload", ns.Name).Queue(lq.Name).Obj()
 			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
 			gomega.Expect(util.SetQuotaReservation(ctx, k8sClient, wl, testing.MakeAdmission(cq.Name).Obj())).To(gomega.Succeed())
+			util.SetWorkloadsAdmissionCkeck(ctx, k8sClient, wl, check.Name, kueue.CheckStateReady, true)
+			gomega.Eventually(func(g gomega.Gomega) {
+				key := client.ObjectKeyFromObject(wl)
+				updatedWl := &kueue.Workload{}
+				g.Expect(k8sClient.Get(ctx, key, updatedWl)).To(gomega.Succeed())
+				g.Expect(apimeta.IsStatusConditionTrue(updatedWl.Status.Conditions, kueue.WorkloadAdmitted)).To(gomega.BeTrue())
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
```

Comment:
```suggestion
			gomega.Eventually(func() bool {
				key := client.ObjectKeyFromObject(wl)
				updatedWl := &kueue.Workload{}
				if errr := k8sClient.Get(ctx, key, updatedWl); err != nil {
					return false
				}
				return apimeta.IsStatusConditionTrue(updatedWl.Status.Conditions, kueue.WorkloadAdmitted)
			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
```

Reply:
`func(g gomega.Gomega)` version gives better assertion reports on failure wjile the both of them are doing the same thing

Reply:
I'm ok with using `func(g gomega.Gomega)`. However, I don't find this way only here.
So, if you believe that this way is much better, I would suggest replacing `func() bool` with this way in all places as a follow up.

This means that I think we should use `func() bool` in this PR.

Reply:
we are using `func(g gomega.Gomega)` in other places already

Reply:
I prefer `func(g gomega.Gomega)`. It will give better error reporting than returning `bool`. I just didn't know this was possible before.

Reply:
> we are using func(g gomega.Gomega) in other places already

I didn't find that this way has already been used. If so, I'm ok.

Reply:
https://github.com/kubernetes-sigs/kueue/blob/25b6dbfe30d04d970e38c0d203003f0acf72d7ed/test/util/util.go#L465-L481

Reply:
Thanks for pointing this out.